File: 프로그래머스/unrated/148653. 마법의 엘리베이터/마법의 엘리베이터.java
import java.util.*;

class Solution {
    public int solution(int NextStorey) {
        int answer = 0;

        while (NextStorey > 0){
            int NowRemain = NextStorey % 10;
            NextStorey /= 10; 
            int nextRest = NextStorey % 10;

            if (NowRemain > 5 && NextStorey == 0) {
                answer += 11 - NowRemain;
            }
            else if (NowRemain > 5) {
                answer += 10 - NowRemain;
                NextStorey += 1;
            }
            else if (NowRemain == 5){
                if (nextRest >= 5) NextStorey += 1;
                answer += 5;
            }
            else {
                answer += NowRemain;
            }
        }

        return answer;
    }
}

File: 프로그래머스/unrated/148653. 마법의 엘리베이터/마법의 엘리베이터.cpp
#include <string>
#include <vector>

using namespace std;
int answer = 100000000;
int f(int num, int stone){
        
    if(num<10){
       stone+=min(num, 11-num);
       answer = min(answer,stone);
       return 0;
    }
    
    
    int l = num%10;
    num/=10;
    if(num>0) f(num +1, stone+10-l);                  
    f(num , stone+l);
    
    return 0;
}
int solution(int storey) {
    
    f(storey, 0);
    return answer;
}